src/services/JobRoleService.ts

```diff
@@ -15,4 +13,20 @@ export const getJobRoles = async (token: string): Promise<JobRoleResponse[]> =>
         console.log(e);
         throw new Error('Failed to get Job Roles');
     }
+}
+
+export const getJobRoleByID = async (id: string, token: string): Promise<JobRoleDetailResponse> =>{
+    if(id == null || id == '')
+        throw new Error ('Invalid Job ID')
+    try {
+        const response: AxiosResponse = await axios.get(URL + id, getHeader(token));
+        return response.data;
+    } catch (e) {
+        if (e.response?.status == 400)
+            throw new Error('Role does not exist')
+        else if (e.response?.status == 500)
+        throw new Error ('Failed to get Job Role Details');
+        else 
+        throw new Error ('Role does not exist');
```

Comment:
I reckon we've some redundancy here. Let's just do a check for 404 and then an else for failed to get. As we've two clauses here giving role does not exist